connect/src/com/telenor/connect/SdkProfile.java

```diff
@@ -24,10 +24,23 @@
     WellKnownAPI.WellKnownConfig getWellKnownConfig();
     boolean isInitialized();
 
-    void onStartAuthorization(OnStartAuthorizationCallback callback);
+    void initializeFromUri(
+            Map<String, String> parameters,
+            List<String> uiLocales,
+            Uri initFrom,
+            OnDeliverAuthorizationUriCallback callback);
 
-    interface OnStartAuthorizationCallback {
-        void onSuccess();
+    void onStartAuthorization(
+            Map<String, String> parameters,
+            List<String> uiLocales,
+            OnStartAuthorizationCallback callback);
+
+    interface OnDeliverAuthorizationUriCallback {
+        void onSuccess(Uri authorizationStartUri);
         void onError();
     }
+
+    interface OnStartAuthorizationCallback extends OnDeliverAuthorizationUriCallback {
```

Comment:
1. They don't have to publish anything if you move the activity into the SDK manifest.

2.  You are putting something that's very generally used in a very low level implementation hook. E.g. let's say they don't use WebViews, but something else? Do you still want your OperatorSelection handling to live there.
2b smells like YAGNI, KISS pls :)